tt_metal/impl/debug/dprint_server.cpp

```diff
@@ -600,6 +664,14 @@ void DebugPrintServerContext::AttachDevice(Device* device) {
         }
     }
 
+    device_intermediate_streams_force_flush_lock_.lock();
```

Comment:
@sagarwalTT I started seeing seeing an assertion on main with a debug build coming from here

```
export TT_METAL_DPRINT_CORES=all
export TT_METAL_DPRINT_ETH_CORES=all
export TT_METAL_DPRINT_CHIPS=0,1
``

Reply:
what is the assertion error you're encountering?

Reply:
Nvm. Something is wrong with the state of my system...